### PR TITLE
Introduce runtime check that it is root isolate that makes UI native calls.

### PR DESCRIPTION
**Pull request by aam**
With upcoming '--enable-isolate-groups' Dart VM option, flutter engine won't be able to register different native call handlers for same library in isolates in the same isolate group. The reason is that library objects are shared between all isolates in the same isolate group.

Proposed solution introduces runtime check into native calls.

### CODE REVIEW

**Comment by aam** [comment]
Hi @jason-simmons , please let me know what you think about this.

File: lib/ui/text/paragraph.cc [hunk 1]
```diff
@@ -52,42 +52,52 @@ size_t Paragraph::GetAllocationSize() {
 }
 
 double Paragraph::width() {
+  UIDartState::ThrowIfUIOperationsProhited();
```
Comment (chinmaygarde) [inline lib/ui/text/paragraph.cc:55]: This seems redundant. Methods invocations on this C++ object should already go through `FOR_EACH_BINDING` to `DART_NATIVE_CALLBACK` which you have already guarded below.

**Comment by aam** [comment]
> why can't we just update `DartLibraryNative::Register` to forego the registration if made on the wrong thread. That way, the callback should be faster as well.

Foregoing of registration won't work because registration is not per isolate. Registration is per-library and in case of enabled isolate groups where isolates share library objects, affects all isolates in a group. So this change effectively makes secondary isolates register same native handlers, which is harmless but redundant(currently secondary isolate registers reduced number of  native handlers breaking root isolate native handler lookups associated with library objects shared between two isolates). 
Secondary isolate spawned in existing isolate group don't really need to register native handlers. As a follow-up we could skip that registration for isolates spawned in existing isolate group. In other words, registration has to be done only when isolate group is created. And then, runtime check will ensure that UI-related native calls are only made from the root isolate.

File: lib/ui/text/paragraph.cc (hunk 1 above)
Comment (aam) [reply lib/ui/text/paragraph.cc]: Ah, true. Removed.

**Comment by aam** [comment]
Thanks Chinmay. cc @mkustermann

**Pull request merged**